pkg/agent/util/net_linux.go

```diff
@@ -376,3 +378,151 @@ func renameHostInterface(oriName string, newName string) error {
+		link, err := netlink.LinkByName(ifaceName)
+		if err != nil {
+			klog.V(4).InfoS("OVS bridge local port is not ready", "port", ifaceName, "err", err)
+			return false, nil
+		}
+		klog.InfoS("OVS bridge local port is ready", "type", link.Type(), "attrs", link.Attrs())
+		return true, nil
+	}); err != nil {
+		return "", false, fmt.Errorf("failed waiting for internal port to show up: %v", err)
+	}
+
+	localLink, err := netlink.LinkByName(ifaceName)
+	if err != nil {
+		return "", false, err
+	}
+	if _, _, err = SetLinkUp(ifaceName); err != nil {
+		return "", false, fmt.Errorf("failed to set link up: %v", err)
+	}
+
+	// Check if interface is configured with an IPv6 address: if it is, we need to ensure that IPv6
+	// is enabled on the OVS internal port as we need to move all IP addresses over.
+	for _, ip := range ifaceIPs {
+		if ip.IP.To4() == nil {
+			klog.InfoS("Interface has IPv6 address, ensuring that IPv6 is enabled on bridge local port", "port", ifaceName)
+			if err := EnsureIPv6EnabledOnInterface(ifaceName); err != nil {
+				klog.ErrorS(err, "Failed to ensure that IPv6 is enabled on bridge local port, moving uplink IPs to bridge is likely to fail", "port", ifaceName)
+			}
+			break
+		}
+	}
+
+	if err = ConfigureLinkAddresses(localLink.Attrs().Index, ifaceIPs); err != nil {
+		return "", false, err
+	}
+	if err = ConfigureLinkAddresses(iface.Index, nil); err != nil {
+		return "", false, err
+	}
+	// Restore the host routes which are lost when moving the network configuration of the
+	// host interface to OVS bridge interface.
+	if err = ConfigureLinkRoutes(localLink, ifaceRoutes); err != nil {
+		return "", false, err
+	}
+	return bridgedName, false, nil
+}
+
+// RestoreHostInterfaceConfiguration restore the configuration from bridge back to host interface, reverting the
+// actions taken in PrepareHostInterfaceConnection.
+func RestoreHostInterfaceConfiguration(brName string, interfaceName string) {
+	klog.V(4).InfoS("Restoring bridge config to host interface")
+	bridgedName := GenerateUplinkInterfaceName(interfaceName)
+	// restore if interface eth0~ exists
+	if exists, err := interfaceExists(bridgedName); err != nil {
+		klog.ErrorS(err, "Failed to check if interface exists", "interface", bridgedName)
+		return
+	} else if exists {
+		// get interface config
+		var interfaceIPs []*net.IPNet
+		var interfaceRoutes []interface{}
+		if exists, err = interfaceExists(interfaceName); err != nil {
+			klog.ErrorS(err, "Failed to check if interface exists", "interface", interfaceName)
+		} else if exists {
+			_, interfaceIPs, interfaceRoutes, err = GetInterfaceConfig(interfaceName)
+			if err != nil {
+				klog.ErrorS(err, "Failed to get interface config", "interface", interfaceName)
+			}
+		}
+		// remove host interface (eth0~) from bridge
+		if err = deleteOVSPort(brName, bridgedName); err != nil {
+			klog.ErrorS(err, "Delete OVS port failed", "port", bridgedName)
```

Comment:
We probably should return here to avoid a state host interface is still connected to bridge, but renamed back eth0.